### PR TITLE
Fix: load blocks style separately for classic themes

### LINKED ISSUE #10546: Per-block styles on classic themes
With #10543, we load an entire chunk that contains all the styles of every block. #10543 is a temporary patch to fix https://github.com/woocommerce/woocommerce/issues/39658, but we have to find a better solution.

[This post](https://make.wordpress.org/core/2021/07/01/block-styles-loading-enhancements-in-wordpress-5-8/) is a good starting point. We should check the current implementation and change it if doesn't match with WordPress Core implementation.

### PR DESCRIPTION
**Pull request by dinhtungdu**
## What

Fixes #10546 

- Revert #10543
- Adjust the block style loading logic for classic theme.

For better review experience, you can review only https://github.com/woocommerce/woocommerce-blocks/pull/10758/files/8d0dddb4821ac4a44cf13ba37e1d2ef1699a1dcd..9ad4b0eb0e10f2a10fd9f9a006952203b5950f90.

## Why

Because we shouldn't load the unnecessary styles.

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

Testing this PR means making sure there are no styling regressions in existing blocks for classic themes.

0. Activate Storefront theme.
1. Create a post or page and add the All Products block. Verify styles are loaded correctly.
2. Visit the page in the frontend and verify styles are loaded correctly in the frontend as well.
3. Repeat steps 1 and 2 with all blocks listed on [this page](https://wordpress.org/plugins/woo-gutenberg-products-block/). Make sure to test each block individually. So, when possible, try with only one block on the page (in some cases, that's not possible, ie: filter blocks, in that case, try with as few blocks as possible on the page). The reason is that we want to make sure each block includes the style dependencies that it needs, so they need to be tested in isolation, otherwise styles from other blocks might leak into other blocks and "help fix issues".
4. Repeat steps above with a block themes. Block themes are excluded from the newly added logic so there shouldn't be any regression with blocks theme as well, but we should ,at least, smoke test with a block theme.

* [ ] Do not include in the Testing Notes 
* [ ] Should be tested by the development team exclusively 

## Screenshots or screencast

N/a

## WooCommerce Visibility

Required:

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [ ] N/A

## Checklist

Required:
* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [ ] This PR is assigned to a milestone.

Conditional:
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

> Fix: load blocks style separately for classic themes.

### CODE REVIEW

**Comment by dinhtungdu** [comment]
Requesting a review from @woocommerce/rubik as this PR affects the style of Cart and Checkout blocks.

File: src/BlockTypes/AbstractBlock.php [hunk 1]
```diff
@@ -210,6 +210,36 @@ protected function register_block_type() {
 		}
 
 		$metadata_path = $this->asset_api->get_block_metadata_path( $this->block_name );
+
+		/**
+		 * We want to load blocks style separatedly, always, for every themes.
+		 * When the core assets are loaded separately, other blocks' styles get
+		 * enqueue separately too. Thus we only need to handle the remaining
+		 * case.
+		 */
```
Comment (opr) [inline src/BlockTypes/AbstractBlock.php]:
```suggestion
		/**
		 * We always want to load block styles separately, for every theme.
		 * When the core assets are loaded separately, other blocks' styles get
		 * enqueued separately too. Thus we only need to handle the remaining
		 * case.
		 */
```

**Review by opr — DISMISSED** [review]
Hi @dinhtungdu looks like styles aren't loading correctly in the editor, I added this block and a few others and noticed it was missing for them all.

image

No errors in the console though. Also I made a suggestion to make the comment a little better.

**Comment by dinhtungdu** [comment]
@opr Thanks for the review and suggestion. I fixed the style loading in the editor, can you please take another look at this?

**Review by opr — APPROVED** [review]
The changes seem to work OK and styles for the blocks are loading fine in the Cart and Checkout, and on Storefront and Twenty Twenty-Three!

Just to note, the build still takes ~40s when changing `assets/js/blocks/cart/block.js.

**Review by danielwrobert — APPROVED** [review]
I tested the [list of our blocks](https://wordpress.org/plugins/woo-gutenberg-products-block/) on Storefront and smoke-tested a good handful of them in TT3 as well.

All of the styles appeared to load as expected in the Editor and on the frontend of both themes.

**Pull request merged**